### ISSUE TITLE
Does this engine support Distributed deployment?

### ISSUE THREAD

**Issue by qingranfjj** [mzheravin/exchange-core#69]
I use this in my application and I have two servers, does this engine support Distributed deployment，if not ,how can I ensure the engine works when one machine down

**Comment by mzheravin** [comment]
Currently there is no such option from the box, but we working on that. Exchange core is designed to be used in distributed environment.

**Comment by chipzzz** [comment]
@qingranfjj , run it on both machines and put a load balancer in front of your 2 machines, thus load balancing requests across your machines.

**Comment by mzheravin** [comment]
@chipzzz it does not work like this :) Exchange core is a stateful service.

**Comment by severos** [comment]
I'm looking into using exchange-core for a proof-of-concept project that might end up being in a distributed system.
After a fast look on the engine, it seems the simplest way to run in a distributed environment is to have each instance to handle a sub-set of the pairs (symbols), and having a gateway in front it to send requests to the responsible server. User data will have to be shared from a single source.

Does this sound correct viable solution?

**Comment by mzheravin** [comment]
@severos yes, distributing symbols between different nodes is the most known technique. 
In exchange-core orderbooks (symbol state) are evenly distributed between different cores. Risk processing is also distributed between few cores, even though this technique does not provide the same parallelization efficiency as symbol distribution, it still gives some boost.
Same idea can be applied to physical machines. However I don't see much sense, as even a single machine has very high throughput.
On a machine level I prefer to have a redundant cluster.
